### PR TITLE
Fix disable execution setting and enable rate limiting

### PR DESCRIPTION
**Pull request by drewbutlerbb4**
Fixes the disable execution option in settings so that execution of pipelines can be disabled generally rather than just for non-admins.

Adds rate limiting to address security vulnerability.

### CODE REVIEW

**Comment by drewbutlerbb4** [comment]
@ckadner 
Hopefully this resolves the security issue. Also gives an additional way to disable execution of assets

**Comment by drewbutlerbb4** [comment]
Signed-off-by: Andrew Butler <Andrew.Butler@ibm.com>

File: dashboard/origin-mlx/server/server.ts [hunk 1]
```diff
@@ -98,6 +99,12 @@ if (REACT_APP_BASE_PATH.length !== 0) {
   app.use('/experiments/', staticHandler);
 }
 
+var limiter = new ratelimit({
+  windowMs: 1*60*1000, // 1 minute
+  max: 5
```
Comment (Tomcli) [inline dashboard/origin-mlx/server/server.ts:104]: 5 requests for 1 minutes might be too little? Remember when running MLX on Kubernetes, all the incoming API requests will also go through this backend for frontend server just like KFP.

Comment (Tomcli) [reply dashboard/origin-mlx/server/server.ts:104]: We allow around 100 requests per minutes.

Comment (ckadner) [reply dashboard/origin-mlx/server/server.ts:104]: we may have to iterate over that value, but I agree that `5` is too few

What is the error message to the user (or in logs) if that limit is exceeded?

**Review by ckadner — CHANGES_REQUESTED** [review]

**Comment by drewbutlerbb4** [comment]
The security issue only exists for the following lines, due to the fact that accessing the file system can be slow.
```
app.get('*', (req, res) => {
  // TODO: look into caching this file to speed up multiple requests.  
  res.sendFile(path.resolve(staticDir, 'index.html'));
})
```
The specific security request is that anytime the file system is accessed there should be some sort of rate limiting. I believe the rate limiter only affects routes after it is initialized and `app.use(rateLimiter)` is run (in our case, this is just when a route is unmatched), but I modified the statement so that the rate limiter is explicitly only used on the `app.get('*',...)` request. This should not affect the API routes at all and should only affect routes which are unmatched which should not happen very often. I can raise the number higher 

The `app.get('*', ...)` route is just a catch all route that we used so that the user could type in almost any address without getting weird results. Like I said in the TLDR I think this is a red herring for bad routing. The server routes to index.html as a catch all for routes it could not match previously. The server should probably just route to a specific 404 page through the static handler rather than routing to index.html through file access. This would avoid the security issue as nothing would be handled by file system access and an added 404 page would be more helpful feedback for end-users, as opposed to a blank UI. I'll address this in a followup PR.

**Comment by drewbutlerbb4** [comment]
Thanks @Tomcli @ckadner, I address your comments above but here is a tldr.

Too Long, Didn't Read:
The server should only rate limit when no route is matched (in the catch-all route). In this case I think it is okay to have a low rate limit, because attempting to go to addresses that can't be matched by the server indicates some form of use that we didn't intend. That being said, I can bump it up to 100 if you think 5 is too low even for that.

More generally, I think this security issue is a red herring for improper routing. After this PR I'll work on a PR to address this.

**Comment by Tomcli** [comment]
> Thanks @Tomcli @ckadner, I address your comments above but here is a tldr.
> 
> Too Long, Didn't Read:
> The server should only rate limit when no route is matched (in the catch-all route). In this case I think it is okay to have a low rate limit, because attempting to go to addresses that can't be matched by the server indicates some form of use that we didn't intend. That being said, I can bump it up to 100 if you think 5 is too low even for that.
> 
> More generally, I think this security issue is a red herring for improper routing. After this PR I'll work on a PR to address this.

thanks for the clarification. If the rate limit is only for the improper routes, then it's ok to keep it as 5 since we are not blocking the actual API calls.

**Comment by Tomcli** [comment]
@drewbutlerbb4 can you sign the dco? thanks

**Comment by drewbutlerbb4** [comment]
Thanks @Tomcli the DCO should be signed now.

@ckadner I forgot to mention but the message displayed to the user when they get rate limited is:
```Too many requests, please try again later.```

**Comment by Tomcli** [comment]
@ckadner any feedbacks before i merge?

**Review by ckadner — APPROVED** [review]
LGTM

**Pull request merged**